staticaddr/loopin/loopin.go

```diff
@@ -0,0 +1,590 @@
+	feeRate chainfee.SatPerKWeight, network *chaincfg.Params,
+	blockHeight uint32, maxFeePercentage float64) (*wire.MsgTx, error) {
+
+	if network == nil {
+		return nil, errors.New("no network provided")
+	}
+
+	htlc, err := l.getHtlc(network)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the sweep transaction.
+	sweepTx := wire.NewMsgTx(2)
+	sweepTx.LockTime = blockHeight
+
+	var weightEstimator input.TxWeightEstimator
+	weightEstimator.AddP2TROutput()
+
+	err = htlc.AddSuccessToEstimator(&weightEstimator)
+	if err != nil {
+		return nil, err
+	}
+
+	htlcTx, err := l.createHtlcTx(
+		network, l.HtlcTxFeeRate, maxFeePercentage,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add the htlc input.
+	sweepTx.AddTxIn(&wire.TxIn{
+		PreviousOutPoint: wire.OutPoint{
+			Hash:  htlcTx.TxHash(),
+			Index: 0,
+		},
+		SignatureScript: htlc.SigScript,
+		Sequence:        htlc.SuccessSequence(),
+	})
+
+	// Add the sweep output.
+	sweepPkScript, err := txscript.PayToAddrScript(sweepAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	fee := feeRate.FeeForWeight(weightEstimator.Weight())
+
+	htlcOutValue := htlcTx.TxOut[0].Value
+	output := &wire.TxOut{
+		Value:    htlcOutValue - int64(fee),
+		PkScript: sweepPkScript,
+	}
+
+	sweepTx.AddTxOut(output)
+
+	signDesc := lndclient.SignDescriptor{
+		WitnessScript: htlc.TimeoutScript(),
+		Output: &wire.TxOut{
+			Value:    htlcOutValue,
+			PkScript: htlc.PkScript,
+		},
+		HashType:   htlc.SigHash(),
+		InputIndex: 0,
+		KeyDesc: keychain.KeyDescriptor{
+			KeyLocator: l.HtlcKeyLocator,
+		},
+	}
+	signDesc.SignMethod = input.WitnessV0SignMethod
+
+	rawSigs, err := signer.SignOutputRaw(
+		ctx, sweepTx, []*lndclient.SignDescriptor{&signDesc}, nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("sign output error: %w", err)
+	}
+	sig := rawSigs[0]
+
+	// Add witness stack to the tx input.
+	sweepTx.TxIn[0].Witness, err = htlc.GenTimeoutWitness(sig)
+	if err != nil {
+		return nil, err
+	}
+
+	return sweepTx, nil
+}
+
+// pubkeyTo33ByteSlice converts a pubkey to a 33 byte slice.
+func pubkeyTo33ByteSlice(pubkey *btcec.PublicKey) [33]byte {
+	var pubkeyBytes [33]byte
+	copy(pubkeyBytes[:], pubkey.SerializeCompressed())
+
+	return pubkeyBytes
+}
+
+// TotalDepositAmount returns the total amount of the deposits.
+func (l *StaticAddressLoopIn) TotalDepositAmount() btcutil.Amount {
+	var total btcutil.Amount
+	if len(l.Deposits) == 0 {
+		return total
+	}
+
+	for _, d := range l.Deposits {
+		total += d.Value
+	}
+	return total
+}
+
+// RemainingPaymentTimeSeconds returns the remaining time in seconds until the
+// payment timeout is reached. The remaining time is calculated from the
+// initiation time of the swap. If more than the swaps configured payment
+// timeout has passed, the remaining time will be negative.
+func (l *StaticAddressLoopIn) RemainingPaymentTimeSeconds() int64 {
+	elapsedSinceInitiation := time.Since(l.InitiationTime).Seconds()
```

Comment:
I propose to add lnd/clock instance to StaticAddressLoopIn and use it here, so this place can be fully tested (by passing a test clock). I propose to add a unit test for this method, testing it at various times.